### PR TITLE
Docs: Policy Syntax page to include UI details.

### PR DESCRIPTION
**Pull request by aphorise**
New section to include details of builtin UI related portions.

Screenshot 2024-02-15 at 19 19 48

Requires further review since this is a new section.

### CODE REVIEW

File: website/content/docs/concepts/policies.mdx [hunk 1]
```diff
@@ -870,6 +870,37 @@ However, the _contents_ of policies are parsed in real-time whenever the token i
 As a result, if a policy is modified, the modified rules will be in force the
 next time a token, with that policy attached, is used to make a call to Vault.
 
+## UI Related
+
+The Vault UI has a few considerations that are different to the capabilities 
+granted to all other CLI or API paths. When the [UI is `enabled`](/vault/docs/configuration/ui#activating-the-vault-ui) 
+and using the `default` policy the following paths are automatically included:
```
Comment (jasonodonnell) [inline website/content/docs/concepts/policies.mdx:877]:
```suggestion
The Vault UI may require different ACL capabilities compared to the CLI. When [enabled](/vault/docs/configuration/ui#activating-the-vault-ui) 
and using the `default` policy the following paths are automatically included:
```

File: website/content/docs/concepts/policies.mdx [hunk 2]
```diff
@@ -870,6 +870,36 @@ However, the _contents_ of policies are parsed in real-time whenever the token i
...
+
+The [/sys/internal/ui/mounts](/vault/api-docs/system/internal-ui-mounts) API provide a 
+list of mounts as per what's typically visible when logging into Vault and observing the 
+personal cubbyhole for example. The listing of mounts are determined by [the `listing_visibility` parameter](/vault/api-docs/system/mounts#listing_visibility) 
+that may be configured during the initial enabling of the mount or otherwise 
+[tuned afterwards](/vault/api-docs/system/auth#tune-auth-method).
+
+The [/sys/internal/ui/resultant-acl](/vault/api-docs/system/internal-ui-resultant-acl) API 
+is required for appropriate adaptation to responses used by the UI and if it were
+inaccessible then you may observe related warning or error messages within the UI itself.
```
Comment (schavis) [inline website/content/docs/concepts/policies.mdx:892]:
```suggestion
Vault uses the [/sys/internal/ui/resultant-acl](/vault/api-docs/system/internal-ui-resultant-acl)
endpoint to repackage information used by the UI. If you do not have have
permission to call the `ui/resultant-acl` endpoint, you may receive warnings or
errors in the UI.
```

I'm not sure I actually understood what this block was trying to say, so may edits may not be entirely accurate.

File: website/content/docs/concepts/policies.mdx [hunk 3]
```diff
@@ -870,6 +870,36 @@ However, the _contents_ of policies are parsed in real-time whenever the token i
...
+
+~> **Note:** [/sys/internal/ui/mounts](/vault/api-docs/system/internal-ui-mounts) is only used 
+internally for UI or CLI preflight checks and it is an unauthenticated API. When a `X-Vault-Token` 
+is presented with requests, the response will include additional mounts which the token has 
+been granted path capabilities on.
+
+!> **Note:** No policy adjustment or overwrite can be made to these UI related path. The 
+default `deny` or other policies targeting these paths with an [explicit `deny`](/vault/docs/concepts/policies#deny)
+capability will have no effect.
```
Comment (schavis) [inline website/content/docs/concepts/policies.mdx:901]:
```suggestion
```

Moved to the beginning of the section

File: website/content/docs/concepts/policies.mdx [hunk 4]
```diff
@@ -870,6 +870,36 @@ However, the _contents_ of policies are parsed in real-time whenever the token i
...
+[tuned afterwards](/vault/api-docs/system/auth#tune-auth-method).
+
+The [/sys/internal/ui/resultant-acl](/vault/api-docs/system/internal-ui-resultant-acl) API 
+is required for appropriate adaptation to responses used by the UI and if it were
+inaccessible then you may observe related warning or error messages within the UI itself.
+
+~> **Note:** [/sys/internal/ui/mounts](/vault/api-docs/system/internal-ui-mounts) is only used 
+internally for UI or CLI preflight checks and it is an unauthenticated API. When a `X-Vault-Token` 
+is presented with requests, the response will include additional mounts which the token has 
+been granted path capabilities on.
```
Comment (schavis) [inline website/content/docs/concepts/policies.mdx:897]:
```suggestion
<Note title="The ui/mounts endpoint is unauthenticated"> 

  Vault uses the endpoint for UI and CLI preflight checks and is intended for
  internal use only. Requests that include an `X-Vault-Token` will return
  all mounts the token has path capabilities on.

</Note>
```

**Review by schavis — CHANGES_REQUESTED** [review]

**Comment by yhyakuna** [comment]
@schavis  I created a [Jira task](https://hashicorp.atlassian.net/browse/SPE-742). Let's convert this into a partial and also ad this to the UI docs.

**Comment by schavis** [comment]
[PR 25956](https://github.com/hashicorp/vault/pull/25956) to add the UI policy info as a partial is in review

File: website/content/docs/concepts/policies.mdx [hunk 5]
```diff
@@ -870,6 +870,46 @@ However, the _contents_ of policies are parsed in real-time whenever the token i
...
+permission to call the `ui/resultant-acl` endpoint, you may receive warnings or
+errors in the UI.
+
+<Note title="The ui/mounts endpoint is unauthenticated"> 
+
+  Vault uses the endpoint for UI and CLI preflight checks and is intended for
+  internal use only. Requests that include an `X-Vault-Token` will return
+  all mounts the token has path capabilities on.
+
+</Note>
```
Comment (schavis) [inline website/content/docs/concepts/policies.mdx:910]:
```suggestion
## UI policy requirements

@include 'ui/policy-requirements.mdx'
```

Updated to use the new partial

**Review by schavis — CHANGES_REQUESTED** [review]

**Review by schavis — APPROVED** [review]

**Pull request merged**